devito/core/gpu_openmp.py

```diff
@@ -2,18 +2,26 @@
 
 import cgen as c
 
+from copy import deepcopy
```

Comment:
this import should be moved under `from functools import ...` as it's not a devito import, but rather an import from the python library

also, I haven't checked yet what you use deepcopy for, but it's 99% wrong